### PR TITLE
Tag and badge for Dart 3 incompatible packages.

### PR DESCRIPTION
**Pull request by isoos**
#6614

### CODE REVIEW

File: app/lib/package/model_properties.dart [hunk 1]
```diff
@@ -135,6 +135,13 @@ class Pubspec {
               .intersect(VersionConstraint.parse('>=2.0.0'))
               .isEmpty);
 
+  late final isDart3Incompatible =
+      !supportsOnlyLegacySdk && // do not mix is:legacy with is:dart3-incompatible
+          _hasDartSdkConstraint &&
+          !_dartSdkConstraint!
+              .intersect(VersionConstraint.parse('<2.12.0'))
```
Comment (jonasfj) [inline app/lib/package/model_properties.dart:142]: I'm guessing this is slightly better:
```suggestion
              .intersect(VersionConstraint.parse('<2.12.0-0'))
```

File: app/lib/tool/test_profile/import_source.dart [hunk 2]
```diff
@@ -129,7 +129,7 @@ class _AutoGeneratedImportSource implements ImportSource {
     final hasRepository = hasher('hasRepository', max: 20) > 0;
     final isLegacy = version.contains('legacy');
 
-    final sdkConstraint = isLegacy ? '>=1.12.0 <2.0.0' : '>=2.6.0 <3.0.0';
+    final sdkConstraint = isLegacy ? '>=1.12.0 <2.0.0' : '>=2.13.0 <3.0.0';
```
Comment (jonasfj) [inline app/lib/tool/test_profile/import_source.dart:132]: It might be wise to do this in a separate PR. Not that I see anything scary in the golden files.

**Review by jonasfj — APPROVED** [review]

File: app/lib/tool/test_profile/import_source.dart (hunk 2 above)
Comment (isoos) [reply app/lib/tool/test_profile/import_source.dart:132]: It will change the golden files either way: if we don't change this, the goldens will show the dart3-incompatible badge.
